server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java

```diff
@@ -65,18 +73,19 @@ public SparseVectorFieldMapper build(MapperBuilderContext context) {
     }
 
     public static final TypeParser PARSER = new TypeParser((n, c) -> {
-        if (c.indexVersionCreated().onOrAfter(IndexVersion.V_8_0_0)) {
-            throw new IllegalArgumentException(ERROR_MESSAGE);
-        } else {
+        if (c.indexVersionCreated().before(IndexVersion.V_8_0_0)) {
             deprecationLogger.warn(DeprecationCategory.MAPPINGS, "sparse_vector", ERROR_MESSAGE_7X);
-            return new Builder(n);
+        } else if (c.indexVersionCreated().before(IndexVersion.V_8_11_0)) {
```

Comment:
I am not completely sure, but I think we should use `IndexVersion.V_8_500_000`?

@thecoop What do you think? If we want to introduce a new field time, should we use a new index version format?

Reply:
Yeah, you should use a new version here. Think about what happens when the current main code is upgraded to this version.

Reply:
Understood - Added a new IndexVersion and used it in the code. Thanks for the feedback!